models/POP/shell_scripts/cesm2_1/perfect_model.csh

```diff
@@ -0,0 +1,147 @@
+set OCN_DATE     = `echo $FILE:e | sed -e "s#-# #g"`
+# Piping the indexed substrings to bc strips away the leading zeros
+set OCN_YEAR     = `echo $OCN_DATE[1] | bc`
+set OCN_MONTH    = `echo $OCN_DATE[2] | bc`
+set OCN_DAY      = `echo $OCN_DATE[3] | bc`
+set OCN_SECONDS  = `echo $OCN_DATE[4] | bc`
+set OCN_HOUR     = `echo $OCN_DATE[4] / 3600 | bc`
+
+echo "valid time of model is $OCN_YEAR $OCN_MONTH $OCN_DAY $OCN_SECONDS (seconds)"
+echo "valid time of model is $OCN_YEAR $OCN_MONTH $OCN_DAY $OCN_HOUR (hours)"
+
+#=========================================================================
+# Block 2: Populate a run-time directory with the obs_seq needed to run DART.
+#=========================================================================
+
+# Get observation sequence file ... or die right away.
+# The observation file names have a time that matches the stopping time of POP.
+
+set YYYYMM   = `printf %04d%02d                ${OCN_YEAR} ${OCN_MONTH}`
+set OBSFNAME = `printf obs_seq.0Z.%04d%02d%02d ${OCN_YEAR} ${OCN_MONTH} ${OCN_DAY}`
+set OBS_FILE = ${BASEOBSDIR}/${YYYYMM}/${OBSFNAME}
+
+if (  -e   ${OBS_FILE} ) then
+   ${LINK} ${OBS_FILE} obs_seq.in
+else
+   echo "ERROR ... no observation file $OBS_FILE"
+   echo "ERROR ... no observation file $OBS_FILE"
+   exit -2
+endif
+
+#=========================================================================
+# Block 3: Populate a run-time directory with the input.nml needed to run DART.
+#=========================================================================
+
+echo "`date` -- BEGIN COPY BLOCK"
+
+if ( -e ${CASEROOT}/input.nml ) then
+   ${COPY} ${CASEROOT}/input.nml .
+else
+   echo "ERROR ... DART required file ${CASEROOT}/input.nml not found ... ERROR"
+   echo "ERROR ... DART required file ${CASEROOT}/input.nml not found ... ERROR"
+   exit -3
+endif
+
+echo "`date` -- END COPY BLOCK"
+
+set OCN_RESTART_FILENAME = ${CASE}.pop.r.${OCN_DATE_EXT}.nc
+# BKJ I think setting this is unnecessary
+# set     OCN_NML_FILENAME = pop_in
+
+${LINK} ./$OCN_RESTART_FILENAME pop.r.nc
+# BKJ I think setting this is unnecessary
+# ${LINK} ../$OCN_NML_FILENAME     pop_in
+
+#=========================================================================
+# Block 4: Advance the model and harvest the synthetic observations.
+# output files are:
+# True_state.nc   ...... the DART state
+# obs_seq.perfect ...... the synthetic observations
+# dart_log.out    ...... run-time output of all DART routines
+#=========================================================================
+
+echo "`date` -- BEGIN POP PERFECT_MODEL_OBS"
+
+${LAUNCHCMD} ${EXEROOT}/perfect_model_obs
+
+if ($status != 0) then
+   echo "ERROR ... DART died in 'perfect_model_obs' ... ERROR"
+   echo "ERROR ... DART died in 'perfect_model_obs' ... ERROR"
+   exit -4
+endif
+
+${MOVE} obs_seq.perfect  ../pop_obs_seq.${OCN_DATE_EXT}.perfect
+${MOVE} dart_log.out     ../pop_dart_log.${OCN_DATE_EXT}.out
```

Comment:
If you don't move the true state file ... you only have the last timestep at the end ... traditionally, the True_State.nc had the entire timeseries that generated all the observations.